packages/web3-eth/src/utils/detect_transaction_type.ts

```diff
@@ -110,43 +114,37 @@ export const defaultTransactionTypeParser: TransactionTypeParser = transaction =
 		return '0x1';
 	}
 
-	// We don't return 0x0 here, because if gasPrice is not
-	// undefined, we still don't know if the network
-	// supports EIP-2718 (https://eips.ethereum.org/EIPS/eip-2718)
-	// and whether we should return undefined for legacy txs,
-	// or type 0x0 for legacy txs post EIP-2718
-	if (!isNullish(tx.gasPrice)) {
-		validateTxTypeAndHandleErrors(transactionType0x0Schema, tx, '0x0');
-	}
-
 	const givenHardfork = tx.hardfork ?? tx.common?.hardfork;
-	// If we don't have a hardfork, then we can't be sure we're post
-	// EIP-2718 where transaction types are available
-	if (givenHardfork === undefined) return undefined;
 
-	const hardforkIndex = Object.keys(HardforksOrdered).indexOf(givenHardfork);
+	if (!isNullish(givenHardfork)) {
+		const hardforkIndex = Object.keys(HardforksOrdered).indexOf(givenHardfork);
 
-	// Unknown hardfork
-	if (hardforkIndex === undefined) return undefined;
+		// givenHardfork is London or later, so EIP-2718 is supported
+		if (hardforkIndex >= Object.keys(HardforksOrdered).indexOf('london'))
+			return !isNullish(tx.gasPrice) ? '0x0' : '0x2';
 
-	// givenHardfork is London or later, so EIP-2718 is supported
-	if (hardforkIndex >= Object.keys(HardforksOrdered).indexOf('london'))
-		return !isNullish(tx.gasPrice) ? '0x0' : '0x2';
+		// givenHardfork is Berlin, tx.accessList is undefined, assume type is 0x0
+		if (hardforkIndex === Object.keys(HardforksOrdered).indexOf('berlin')) return '0x0';
+	}
 
-	// givenHardfork is Berlin, tx.accessList is undefined, assume type is 0x0
-	if (hardforkIndex === Object.keys(HardforksOrdered).indexOf('berlin')) return '0x0';
+	const block = await getBlock(web3Context, web3Context.defaultBlock, false, ETH_DATA_FORMAT);
```

Comment:
Tx obj params provided by user should be enough to decide what tx type user want to set, 
if user haven't specified we will default to whatever set in web3.context , and I think NW should not be queried for this decision making here. so i'll suggest to remove getblock.